.eleventy.js

```diff
@@ -35,6 +37,18 @@ module.exports = function (config) {
     return striptags(content);
   });
 
+  config.addFilter("prepTOCContent", (content) => {
+    const toc = config.getFilter("toc");
+    const strip = config.getFilter("strip");
+    const strip_newlines = config.getFilter("strip_newlines")
+    return strip_newlines(strip(toc(content)));
+  });
+
+  config.addFilter("shouldShowTOC", (content, showToc) => {
+    return !!showToc && content !== "" && content !== '<ol id="toc" class="section-nav"></ol>';
```

Comment:
For every new function/filter you add to the eleventy config, do me a solid and comment what it's for. I can kinda tell but it's always nice to have a little more what/why <3

Reply:
In another comment on the docs.html page, you asked for the template markup to be simpler and so I moved the logic to here.  The new docs.html code looks like this now:

                {% assign toc_content = content | prepTOCContent %}
                {% if toc_content | shouldShowTOC: showtoc %}
                <h2>Table of Contents</h2>
                <div id="table-of-contents">
                    {{ toc_content }}
                </div>
                {% endif %}
                
The markup is simpler now, but the logic still needs to exist somewhere.

Reply:
If you're going to add custom functions/filters (magic) to eleventy, it needs to be commented in place for other people to understand what it is and how it works. Maybe you thought I meant here on the PR?

Reply:
I did both so it's tracked both in the code and on the github discussion.